pkg/controller/sshd/sshd_controller.go

```diff
@@ -324,15 +326,28 @@ func (r *ReconcileSSHD) Reconcile(request reconcile.Request) (reconcile.Result,
 			return reconcile.Result{}, err
 		}
 	} else {
-		// Service exists, check if it's updated.
+		var serviceNeedsUpdate bool
+
+		// Service exists, check if annotations or spec need updated.
+
+		if !metav1.HasAnnotation(foundService.ObjectMeta, ELBAnnotationKey) ||
+			foundService.ObjectMeta.Annotations[ELBAnnotationKey] != ELBAnnotationValue {
+			r.SetSSHDStatusPending(instance, "Updating service annotations")
+			metav1.SetMetaDataAnnotation(&foundService.ObjectMeta, ELBAnnotationKey, ELBAnnotationValue)
+			serviceNeedsUpdate = true
+		}
+
 		// XXX Copy system-assigned fields to satisfy reflect.DeepEqual.
 		service.Spec.Ports[0].NodePort = foundService.Spec.Ports[0].NodePort
 		service.Spec.ClusterIP = foundService.Spec.ClusterIP
 		service.Spec.HealthCheckNodePort = foundService.Spec.HealthCheckNodePort
 		if !reflect.DeepEqual(foundService.Spec, service.Spec) {
-			// Specs aren't equal, update and fix.
 			r.SetSSHDStatusPending(instance, "Updating service", "from", foundService.Spec, "to", service.Spec)
 			foundService.Spec = *service.Spec.DeepCopy()
+			serviceNeedsUpdate = true
+		}
+
+		if serviceNeedsUpdate {
 			if err = r.client.Update(context.TODO(), foundService); err != nil {
```

Comment:
Is this needed? If the `reflect.DeepEqual` is false, then doesn't it by definition need an update?

Reply:
It does, but a missing annotation (just above) also sets `serviceNeedsUpdate = true`.

Reply:
I see what's happening now. Note to past me: Don't review code when tired!